Implement fetch likes in PostService

bitdc opened the pull request:
Refs #15662 

Add implementation to fetch likes for a post in `PostService`. 

To test:

Confirm the app builds. The rest are covered through unit tests as this is a "backend" change.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

bitdc committed:
Update source for WordPressKit

Temporarily use WordPressKit from
feature/15662-notification_likes_support which contains fetch likes
capability.

bitdc committed:
Implement fetch post likes in PostService

bitdc committed:
Add tests for fetch post likes

ScoutHarris commented on WordPress/Classes/Services/PostService.m:
```diff
@@ -676,6 +676,26 @@ - (void)restoreRemotePostWithPost:(AbstractPost*)post
     [remote restorePost:remotePost success:successBlock failure:failureBlock];
 }
 
+- (void)getLikesForPost:(AbstractPost *)post
```
I don't know how you're going to call this method, but I wonder if this should be postID instead of AbstractPost. NotificationsVC and NotificationsDetailVC get the PostID from the notification. So it looks like you'd have to create an AbstractPost just to fetch likes. But that's not really necessary since fetching only requires the ID.

bitdc replied:
Ah yes, I was actually in two minds on fetching the likes through `AbstractPost` or simply `postID` and `siteID`. My initial thought was, using `AbstractPost` would reuse most of the existing logic like `PostServiceRemoteFactory` that accepts a `Blog` object, and also makes use of the existing mocking functionalities on the unit testing side. 

As for how the caller utilizes this method, I assume the app would synchronize blogs when the app is opened; and information for the `Blog` and `AbstractPost` can be obtained through `BlogService`:
```objc
- (nullable Blog *)blogByBlogId:(NSNumber *)blogID;
``` 
and `PostService` respectively: 
```objc
- (nullable AbstractPost *)findPostWithID:(NSNumber *)postID inBlog:(Blog *)blog;
```

However, after giving it a second thought, going with just `postID` and `siteID` will make the implementation more reusable should we decide to implement the Likes page from the Reader page in the future – since Reader utilizes a different model (i.e. `ReaderPost`). 

Thanks for your input @ScoutHarris, and I'll make the necessary adjustments shortly! 🙂

bitdc committed:
Update PostService to fetch likes based on ID

Fetching post likes now require postID and siteID, instead of an
AbstractPost instance. Also added methods in `PostServiceRemoteFactory`
to generate a REST service for a given site ID.

Resolves https://github.com/wordpress-mobile/WordPress-iOS/pull/15977#discussion_r583969267.

bitdc committed:
Update tests for fetching post likes

Make minor adjustments since the method signature changed to postID and
siteID, instead of AbstractPost instance.

bitdc committed:
Merge branch 'feature/15662-notification_likes' into task/post-fetch-likes

bitdc committed:
Fix conflicts

bitdc committed:
Remove previous fetch likes implementation

bitdc committed:
Update source for WordPressKit

Temporarily use WordPressKit from
feature/15662-notification_likes_support which contains fetch likes
capability.

bitdc committed:
Implement fetch comment likes in CommentService

bitdc committed:
Add remote factory for CommentService

This enables dependency injection from the unit test side and makes it
easier to test.

bitdc committed:
Add tests for CommentService

ScoutHarris commented:
Hey David.

This branch isn't building. It's failing looking for `CommentStatusFilter`, which was recently added to WordPressKit. Maybe your WPKit `feature/15662-notification_likes_support` is not up to date with our WPKit `feature/15662-notification_likes_support`?

Also, when I installed pods in this branch, I had a local change to `Podfile.lock`. You'll want to make sure this is up-to-date in your branch.

bitdc committed:
Update Podfile.lock

bitdc commented:
I've updated the `Podfile.lock`. Thanks for notifying!

ScoutHarris commented on WordPress/Classes/Services/PostServiceRemoteFactory.swift:
```diff
@@ -16,4 +16,36 @@ import WordPressKit
+    private func apiForRESTRequest(using context: NSManagedObjectContext) -> WordPressComRestApi? {
+        let accountService = AccountService(managedObjectContext: context)
+
+        guard let account = accountService.defaultWordPressComAccount(),
+              let api = account.wordPressComRestApi else {
```
This could just be one line:
`guard let api = accountService.defaultWordPressComAccount()?.wordPressComRestApi`

bitdc replied:
Noted!

ScoutHarris reviewed (approved):
I just had one small suggestion. Otherwise, I think this will work.

Let me know what you decide on the suggestion, and I'll merge this accordingly.

ScoutHarris committed:
Merge branch 'task/comment-fetch-likes' into task/post-fetch-likes

ScoutHarris commented:
Hey David. I'll go ahead and merge this so you can proceed. Please make a note of the suggestion for your next PR. Thanks!
